File: foodcatalogue/src/main/java/com/codedecode/foodcatalogue/entity/FoodItem.java
package com.codedecode.foodcatalogue.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class FoodItem {

    @Id
    @GeneratedValue(strategy = jakarta.persistence.GenerationType.AUTO)
    private int id;
    private String itemName;
    private String itemDescription;
    private boolean isVeg;
    private Number price;
    private Integer restaurantId;

    @Column(nullable = false, columnDefinition = "integer default 0")
    private Integer quantity;
}

File: RestaurantUI/src/app/model/class/FoodItem.ts
export class FoodItem {
    id: number;
    itemName: string;
    itemDescription: string;
    isVeg: boolean;
    price: number;
    restaurantId: number;
    quantity: number;

    constructor(data?: any) {
        if (data) {
        this.id = data.id || 0;
        this.itemName = data.itemName || '';
        this.itemDescription = data.itemDescription || '';
        this.isVeg = data.isVeg !== undefined ? data.isVeg : true;
        this.price = data.price || 0;
        this.restaurantId = data.restaurantId || 0;
        this.quantity = data.quantity || 1;
        } else {
        this.id = 0;
        this.itemName = '';
        this.itemDescription = '';
        this.isVeg = true;
        this.price = 0;
        this.restaurantId = 0;
        this.quantity = 1;
    }
}
}
